controllers/territory_controller/territory_controller.py

```diff
@@ -150,7 +150,12 @@ def claim_territory(
             return
 
         new_colour = ZONE_COLOURS[claimed_by]
-        self._robot.getFromDef(station_code).getField("zoneColour").setSFColor(
+        station = self._robot.getFromDef(station_code)
+        if station is None:
+            raise RuntimeError(
+                f"Failed to fetch territory node {station_code}",
+            )
```

Comment:
Maybe make this just a `logging.error`? If this controller crashes out that won't kill the simulation and the error would just as easily be lost in the logs.

Not sure if we think this should be enough to reject a competition match. If it should, then we should maybe consider putting something in place to force that.